### PR TITLE
Pt 166927306 names as strings

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/166927306

### CODE REVIEW

**Comment by gorillainduction** [comment]
Holdning this merge until the release of 3.2 of the compiler is done.

File: src/aeb_fate_data.erl [hunk 1]
```diff
@@ -208,17 +208,15 @@ ordinal(T) when ?IS_FATE_BOOLEAN(T)   -> 1;
...
-ordinal(T) when ?IS_FATE_MAP(T)       -> 12;
-ordinal(T) when ?IS_FATE_LIST(T)      -> 13;
-ordinal(T) when ?IS_FATE_VARIANT(T)   -> 14;
-ordinal(T) when ?IS_FATE_ORACLE_Q(T)  -> 15.
+ordinal(T) when ?IS_FATE_BITS(T)      -> 8;
+ordinal(T) when ?IS_FATE_STRING(T)    -> 9;
+ordinal(T) when ?IS_FATE_TUPLE(T)     -> 10;
+ordinal(T) when ?IS_FATE_MAP(T)       -> 11;
+ordinal(T) when ?IS_FATE_LIST(T)      -> 12;
+ordinal(T) when ?IS_FATE_VARIANT(T)   -> 13.
```
Comment (UlfNorell) [inline src/aeb_fate_data.erl]: The reason to not renumber these is that the numbers are also used in `lt` below.

Comment (happi) [reply src/aeb_fate_data.erl]: If the FATE_LIST_VALUE match is replaced by IS_FATE_LIST then the number could be removed from lt.

Comment (happi) [reply src/aeb_fate_data.erl]: And the FATE_STRING.

Comment (happi) [reply src/aeb_fate_data.erl]: This is only tested in the quick check tests. Perhaps leave the renaming and I can do it in a separate PR also adding tests. https://www.pivotaltracker.com/story/show/166975637

**Comment by gorillainduction** [comment]
What about this for the ordinals?

**Comment by happi** [comment]
Better.
I noticed that bytes and strings now compare differently internally. Hashes and signatures had fixed sizes but bytes have a size. For strings we first compare the size and then the binary (in Erlang  <<1, 1>> < <<2>> but in fate "AA" > "B"). Bytes should work the same way.

**Comment by UlfNorell** [comment]
> For strings we first compare the size and then the binary

Why do we do this though? The Erlang comparison is simpler.

**Comment by happi** [comment]
@ThomasAerts had some rationale.

**Comment by UlfNorell** [comment]
I'm sure it was it was a valid reason and not based on troll logic.

**Comment by UlfNorell** [comment]
Anyway, in type correct code you never compare byte arrays of different sizes, so comparing the sizes is extra redundant.

**Pull request merged**